src/shell/commands/executable.rs

```diff
@@ -55,22 +57,33 @@ impl ShellCommand for ExecutableCommand {
       // avoid deadlock since this is holding onto the pipes
       drop(sub_command);
 
-      tokio::select! {
-        result = child.wait() => match result {
-          Ok(status) => ExecuteResult::Continue(
-            status.code().unwrap_or(1),
-            Vec::new(),
-            Vec::new(),
-          ),
-          Err(err) => {
-            let _ = stderr.write_line(&format!("{}", err));
-            ExecuteResult::Continue(1, Vec::new(), Vec::new())
-          }
-        },
-        _ = context.state.token().cancelled() => {
+      let child_clone = Arc::clone(&child);
+
+      tokio::spawn(async move {
+        tokio::signal::ctrl_c().await.unwrap();
+        let mut child = child_clone.lock().await;
```

Comment:
I'm not sure how this works because acquiring the lock in one place will block the other.

Instead of spawning a task, I feel like it should work to just add this to the select below? Something like:

```rs
      tokio::select! {
          result = child.wait() => /* ... */,
          _ = tokio::signal::ctrl_c() => {
              let _ = child.kill().await;
              // ...
          }
          _ = context.state.token().cancelled() => {
              let _ = child.kill().await;
              ExecuteResult::for_cancellation()
          }
      }
```

By the way, in the future when opening a PR, make sure to do it from a branch instead of the main branch. There's a couple reasons to do this:

1. It makes it easy to merge the latest main into your fork because your main branch is clean.
2. It allows maintaininers to push to your branch. Right now I get:
    ```
    To https://github.com/mo/deno_task_shell.git
     ! [remote rejected] mo/main -> mo/main (permission denied)
    error: failed to push some refs to 'https://github.com/mo/deno_task_shell.git'
    ```